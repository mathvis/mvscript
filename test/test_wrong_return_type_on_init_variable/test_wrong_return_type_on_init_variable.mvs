func a() int {
	return 0;
}
let b: bool = a();
