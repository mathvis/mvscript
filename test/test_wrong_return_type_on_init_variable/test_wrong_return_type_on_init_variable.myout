Decl (FunctionDef (FunctionIdentifier "a") [] IntT (Just (Block (FunctionBlock IntT) [])))
error: Expected BoolT but got IntT
 --> test/test_wrong_return_type_on_init_variable/test_wrong_return_type_on_init_variable.mvs:2:1
  |
2 | let b: bool = a();
  | ^
  |
hint: Consider changing the variable type or declaring a new variable.
