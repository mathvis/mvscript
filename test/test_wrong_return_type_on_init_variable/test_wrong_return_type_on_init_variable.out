Decl (FunctionDef (FunctionIdentifier "a") [] IntT (Just (Block (FunctionBlock IntT) [Expr (Return (Just (Type (Int 0))))])))
error: Expected BoolT but got IntT
 --> test/test_wrong_return_type_on_init_variable/test_wrong_return_type_on_init_variable.mvs:4:1
  |
4 | let b: bool = a();
  | ^
  |
hint: Consider changing the variable type or declaring a new variable.
